packages/app/src/cli/services/dev/processes/setup-dev-processes.ts

```diff
@@ -80,7 +80,11 @@ export async function setupDevProcesses({
   const appPreviewUrl = buildAppURLForWeb(storeFqdn, apiKey)
   const env = getEnvironmentVariables()
   const shouldRenderGraphiQL = !isTruthy(env[environmentVariableNames.disableGraphiQLExplorer])
-  const shouldPerformAppLogPolling = isTruthy(env[environmentVariableNames.enableAppLogPolling])
+  const shouldPerformAppLogPolling =
+    isTruthy(env[environmentVariableNames.enableAppLogPolling]) &&
+    localApp.allExtensions.filter((extension) => {
+      return extension.specification.identifier === 'function'
+    }).length > 0
```

Comment:
Nit: I think we can use `isFunctionExtension` here 😄 

```suggestion
    localApp.allExtensions.some((extension) => extension.isFunctionExtension)
```